lib/tasks/govuk_publishing_components_tasks.rake

```diff
@@ -1,4 +1,20 @@
-# desc "Explaining what the task does"
-# task :govuk_publishing_components do
-#   # Task goes here
-# end
+desc 'warns if component view files exist without corresponding documentation'
+task :validate_component_documentation do
+  components_missing_docs = []
+  component_views = Dir["app/views/#{GovukPublishingComponents::Config.component_directory_name}/**/*.html.erb"]
+
+  component_views.each do |partial|
+    expected_component_docs_file = partial.split('/')[-1].gsub('html.erb', 'yml').sub("_", "")
+    expected_component_docs_path = "app/views/#{GovukPublishingComponents::Config.component_directory_name}/docs/" + expected_component_docs_file
+    components_missing_docs << partial unless File.exist?(expected_component_docs_path)
+  end
+
+  unless components_missing_docs.empty?
+    error = "\nYou have components which are missing documentation. These components will not be displayed in the component guide:\n"
+    components_missing_docs.each { |component| error += "\t" + component + "\n" }
+    error += "\n"
+    raise error
```

Comment:
Maybe this could be a NotImplementedError with that message?
https://ruby-doc.org/core-2.4.0/NotImplementedError.html

eg https://github.com/alphagov/government-frontend/blob/master/test/component_test_helper.rb#L7